setup.py

```diff
@@ -1,34 +1,29 @@
 import os
-
-from skbuild import setup  # Use scikit-build
-
-# import torch
-# from torch.utils import cpp_extension
+from skbuild import setup
 
 pwd = os.path.abspath(os.path.dirname(__file__))
 with open(os.path.join(pwd, "README.md"), encoding="utf-8") as fp:
     readme_text = fp.read()
 
 # C++ config
-cmake_args = [
-    "-DWITH_PYTHON=1",
-    # f"-DCMAKE_PREFIX_PATH='{os.path.dirname(torch.__file__)};{torch.utils.cmake_prefix_path}'",
-    # f"-DCMAKE_CXX_FLAGS='-D_GLIBCXX_USE_CXX11_ABI={1 if torch._C._GLIBCXX_USE_CXX11_ABI else 0} -fPIC'",
-]
+cmake_args = ["-DWITH_PYTHON=1"]
+
+# Activate CUDA, if found
+try:
+    import torch
+    from torch.utils import cpp_extension
```

Comment:
Should we be using torch for this? There may be JAX users without pytorch (on systems that do not support it) that want to use the CUDA support.

https://mmcv.readthedocs.io/en/v1.4.1/_modules/torch/utils/cpp_extension.html
```
def _find_cuda_home() -> Optional[str]:
    r'''Finds the CUDA install path.'''
    # Guess #1
    cuda_home = os.environ.get('CUDA_HOME') or os.environ.get('CUDA_PATH')
    if cuda_home is None:
        # Guess #2
        try:
            which = 'where' if IS_WINDOWS else 'which'
            with open(os.devnull, 'w') as devnull:
                nvcc = subprocess.check_output([which, 'nvcc'],
                                               stderr=devnull).decode(*SUBPROCESS_DECODE_ARGS).rstrip('\r\n')
                cuda_home = os.path.dirname(os.path.dirname(nvcc))
        except Exception:
            # Guess #3
            if IS_WINDOWS:
                cuda_homes = glob.glob(
                    'C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v*.*')
                if len(cuda_homes) == 0:
                    cuda_home = ''
                else:
                    cuda_home = cuda_homes[0]
            else:
                cuda_home = '/usr/local/cuda'
            if not os.path.exists(cuda_home):
                cuda_home = None
    if cuda_home and not torch.cuda.is_available():
        print(f"No CUDA runtime is found, using CUDA_HOME='{cuda_home}'")
    return cuda_home
```

Reply:
Excellent point. Long-term: no, probably not.
Currently, I simply wanted to fetch all the necessary CUDA flags/options quickly. If we can just steal the CUDA code directly from torch, that would be great. I don't think it's a big problem that we're relying on torch at the moment since that the only CUDA backend we have. If we add JAX, then the above cleare becomes an issue...